test/functional/mongo_client_tests.js

```diff
@@ -734,6 +739,109 @@ describe('MongoClient', function() {
+      uri: 'readPreference=secondary&readpreference=primary',
+      useNewUrlParser: false,
+      expectedReadPreference: 'secondary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readPreference and readpreference) from uri with new parser',
+      uri: 'readPreference=primary&readpreference=secondary',
+      useNewUrlParser: true,
+      expectedReadPreference: 'secondary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readpreference and readPreference) from uri with old parser',
+      uri: 'readpreference=secondary&readPreference=primary',
+      useNewUrlParser: false,
+      expectedReadPreference: 'primary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readpreference and readPreference) from uri with new parser',
+      uri: 'readpreference=primary&readPreference=secondary',
+      useNewUrlParser: true,
+      expectedReadPreference: 'secondary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readPreference and readPreference) from uri with old parser',
+      uri: 'readPreference=secondary&readPreference=primary',
+      useNewUrlParser: false,
+      expectedReadPreference: 'primary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readPreference and readPreference) from uri with new parser',
+      uri: 'readPreference=primary&readPreference=secondary',
+      useNewUrlParser: true,
+      expectedReadPreference: 'secondary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readpreference and readpreference) from uri with old parser',
+      uri: 'readpreference=primary&readpreference=secondary',
+      useNewUrlParser: false,
+      expectedReadPreference: 'primary'
+    },
+    {
+      description:
+        'Should handle duplicate options (readpreference and readpreference) from uri with new parser',
+      uri: 'readpreference=primary&readpreference=secondary',
+      useNewUrlParser: true,
+      expectedReadPreference: 'secondary'
+    },
+    {
+      description: 'Should handle duplicate identical options from uri with new parser',
+      uri: 'readpreference=secondary&readpreference=secondary',
+      useNewUrlParser: true,
+      expectedReadPreference: 'secondary'
+    },
+    {
+      description: 'Should handle three duplicate options from uri with new parser',
+      uri: 'readPreference=primary&readpreference=secondary&readPreference=secondaryPreferred',
+      useNewUrlParser: true,
+      expectedReadPreference: 'secondaryPreferred'
+    }
+  ].forEach(test => {
+    it(test.description, function(done) {
+      const configuration = this.configuration;
+      const url = `${configuration.url()}?${test.uri}`;
+      const client = configuration.newClient(url, { useNewUrlParser: test.useNewUrlParser });
+      let stub;
+
+      if (test.useNewUrlParser) {
+        stub = sinon.stub(console, 'warn');
```

Comment:
It would probably be a good idea to use a [`sinon.sandbox`](https://sinonjs.org/releases/v7.2.3/sandbox/) here.

Reply:
nice, I didn't know about this!